examples/generic_republish.ipynb

```diff
@@ -0,0 +1,283 @@
+   "metadata": {
+    "editable": true,
+    "jp-MarkdownHeadingCollapsed": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Setup\n",
+    "\n",
+    "See \"Hello OpenAssetIO\" notebook for details on how to bootstrap OpenAssetIO. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "df703c44ad53b21c",
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-03-13T08:15:00.152634Z",
+     "start_time": "2024-03-13T08:15:00.141508Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from resources import helpers\n",
+    "\n",
+    "manager, context = helpers.bootstrap(\"resources/generic_republish/openassetio_config.toml\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a80594a-b171-49c5-8cfd-00e334759b1e",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Getting started"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "In the following example we're going to ask the manager about the traits it supports with respect to an asset, and use that to determine a publishing workflow, without knowing anything about the entity.\n",
+    "\n",
+    "We've been given a URI by a colleague, which we need to turn into an `EntityReference` before we can use it to query the asset management system."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "22277554-b477-4600-8c2a-90ee8d5c064e"
+  },
+  {
+   "cell_type": "code",
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": "> **Result:**\n> `<openassetio.EntityReference bal:///project_artwork/logos/openassetio>`"
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "logo_ref = manager.createEntityReference(\"bal:///project_artwork/logos/openassetio\")\n",
+    "\n",
+    "helpers.display_result(repr(logo_ref))"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-03-13T08:15:00.193600Z",
+     "start_time": "2024-03-13T08:15:00.188399Z"
+    }
+   },
+   "id": "82c3962a-27b5-4375-b4fa-dbd14dcfc93b",
+   "execution_count": 9
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Generic resolve and re-publish\n",
+    "\n",
+    "A rare use-case is for a tool that blindly resolves, updates, and re-publishes an entity, regardless what type of entity it is. \n",
+    "\n",
+    "Such generic re-publishing is discouraged and dangerous. For example, if \"approval\" status is blindly resolved and re-published, the manager may not know how it should handle this properly. A better approach to such a generic tool is to present the traits and their properties to the user to select before re-publishing.\n",
+    "\n",
+    "However, such workflows are possible and have their place in a pipeline. This can be accomplished by making use of `entityTraits`. "
```

Comment:
[Consider] Should this section have a code snippet, or is it a prelude to the next section, seem unclear to me.

 I wonder if this section should come last, as we're sort of warning people away. Maybe explicitly call out that an example isn't provided, (because we don't think it's a good idea?)